src/Modules/SimplCommerce.Module.ShoppingCart/Services/CartService.cs

```diff
@@ -136,6 +136,16 @@ public async Task<CouponValidationResult> ApplyCoupon(long userId, string coupon
             return couponValidationResult;
         }
 
+        public async Task SaveOrderNote(long userId, string orderNote)
+        {
+            var cart = _cartRepository.Query().FirstOrDefault(x => x.UserId == userId && x.IsActive);
+            if (cart != null)
```

Comment:
The cart is expected not to be null here. If it's null there must be something wrong here, so I think we better throw an exception if it null

Reply:
Throwing ApplicationException if the Cart is Empty